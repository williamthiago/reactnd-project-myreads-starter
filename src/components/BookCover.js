import React, { Component } from 'react'
import { BookCoverPropType } from '../validations/props'

const detectImageSize = (imageURL, maxHeight) => new Promise(resolve => {
  const image = new Image()

  image.onload = () => {
    let { width, height } = image

    if (height > maxHeight) {
      width *= maxHeight / height
      height = maxHeight
    }

    resolve({ width, height })
  }

  image.src = imageURL
})

class BookCover extends Component {
  state = {
    image: '',
    width: 0,
    height: 0
  }

  async componentDidMount() {
    const maxHeight = 200
    let { image } = this.props
    let size = await detectImageSize(image, maxHeight)
    this.setState({
      ...size,
      image
    })
  }

  render() {
    let { image, width, height } = this.state
    let backgroundSize = 'cover'
    let backgroundImage = `url(${image})`

    const coverStyle = {
      backgroundSize,
      backgroundImage,
      height,
      width,
    }

    return (
      <div className="book-cover" style={coverStyle}></div>
    )
  }
}


BookCover.propTypes = BookCoverPropType

export default BookCover