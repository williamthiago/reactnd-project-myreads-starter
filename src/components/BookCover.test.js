import React from 'react'
import { shallow } from 'enzyme'

import BookCover from './BookCover'

it('Renders BookCover without crashing', () => {
  expect(shallow(<BookCover image="" />)).toMatchSnapshot()
})