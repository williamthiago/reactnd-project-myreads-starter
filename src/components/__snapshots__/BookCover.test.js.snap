// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders BookCover without crashing 1`] = `
<div
  className="book-cover"
  style={
    Object {
      "backgroundImage": "url()",
      "backgroundSize": "cover",
      "height": 0,
      "width": 0,
    }
  }
/>
`;
